pkg/buildpack/buildpack_test.go

```diff
@@ -511,6 +512,38 @@ version = "1.2.3"
 				h.AssertNil(t, err)
 			})
 		})
+
+		when("hardlink is present", func() {
+			var bpRootFolder string
+
+			it.Before(func() {
+				bpRootFolder = filepath.Join("testdata", "buildpack-with-hardlink")
+				// create a hard link
+				err := os.Link(filepath.Join(bpRootFolder, "original-file"), filepath.Join(bpRootFolder, "original-file-2"))
+				h.AssertNil(t, err)
+			})
+
+			it.After(func() {
+				os.RemoveAll(filepath.Join(bpRootFolder, "original-file-2"))
+			})
+
+			it.Focus("hardlink is preserved in the output tar file", func() {
+				bp, err := buildpack.FromBuildpackRootBlob(
+					blob.NewBlob(bpRootFolder),
+					archive.DefaultTarWriterFactory(),
+				)
+				h.AssertNil(t, err)
+
+				tarPath := writeBlobToFile(bp)
+				defer os.Remove(tarPath)
+
+				h.AssertOnTarEntries(t, tarPath,
+					"/cnb/buildpacks/bp.one/1.2.3/original-file",
+					"/cnb/buildpacks/bp.one/1.2.3/original-file-2",
+					h.HardLinks(),
```

Comment:
```suggestion
					h.AreEquivalentHardLinks(),
```
(nit) this more descriptive name could help the reader